dataset/src/main/scala/frameless/TypedColumn.scala

```diff
@@ -15,21 +15,28 @@ sealed class TypedColumn[T, U](
 )(
   implicit
   val uencoder: TypedEncoder[U]
-) extends UntypedExpression[T] {
+) extends UntypedExpression[T] { self =>
 
   def this(column: Column)(implicit uencoder: TypedEncoder[U]) {
     this(FramelessInternals.expr(column))
   }
 
   def untyped: Column = new Column(expr)
 
-  def ===(other: U): TypedColumn[T, Boolean] = {
-    new TypedColumn[T, Boolean](untyped === other)
-  }
+  def ===(other: U): TypedColumn[T, Boolean] = new TypedColumn[T, Boolean](self.untyped === other)
+  def ===(other: TypedColumn[T, U]): TypedColumn[T, Boolean] = new TypedColumn[T, Boolean](self.untyped === other.untyped)
 
-  def ===(other: TypedColumn[T, U]): TypedColumn[T, Boolean] = {
-    new TypedColumn[T, Boolean](untyped === other.untyped)
-  }
+  def +(u: TypedColumn[T, U])(implicit n: Numeric[U]):  TypedColumn[T, U]= new TypedColumn[T, U](self.untyped.plus(u.untyped))
```

Comment:
Agree with @OlivierBlanvillain, not everything that Scala can sum is summable by Spark.